### PR TITLE
ADCM-1624 Add hosts lock check when hostprovider action is running

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: tests/functional/test_locked_objects.py [hunk 1]
```diff
@@ -9,223 +9,195 @@
...
+def test_host_and_hostprovider_must_be_unlocked_when_his_task_finished(
+    hostprovider: Provider, host: Host
+):
+    with allure.step("Run action: action locker and create hostprovider"):
+        task = hostprovider.action_run(name="action-locker")
+    with allure.step("Check if provider and host is locked and then created"):
+        hostprovider.reread()
+        host.reread()
+        assert hostprovider.state == "locked"
+        assert hostprovider.state == "locked"
```
Comment (dgusakov) [inline tests/functional/test_locked_objects.py:198]: Double hostprovider state assertion

File: tests/functional/test_locked_objects.py [hunk 2]
```diff
@@ -9,223 +9,195 @@
...
+    with allure.step("Check if service is locked and then created"):
+        added_service.reread()
+        assert added_service.state == "locked"
+        task.wait()
+        added_service.reread()
+        assert added_service.state == "created"
+
+
+def test_hostprovider_must_be_unlocked_when_his_task_finished(hostprovider: Provider):
+    with allure.step("Run action: action locker and create hostprovider"):
```
Comment (dgusakov) [inline tests/functional/test_locked_objects.py:179]: Looks like typo in allure step
```suggestion
    with allure.step("Run action: action-locker for hostprovider"):
```

File: tests/functional/test_locked_objects.py [hunk 3]
```diff
@@ -9,223 +9,195 @@
...
+        assert hostprovider.state == "locked"
+        task.wait()
+        hostprovider.reread()
+        assert hostprovider.state == "created"
+
+
+def test_host_and_hostprovider_must_be_unlocked_when_his_task_finished(
+    hostprovider: Provider, host: Host
+):
+    with allure.step("Run action: action locker and create hostprovider"):
```
Comment (dgusakov) [inline tests/functional/test_locked_objects.py:192]: Looks like typo in allure step
```suggestion
    with allure.step("Check if provider state is 'locked' and then is 'created'"):
```

File: tests/functional/test_locked_objects.py [hunk 4]
```diff
@@ -9,223 +9,195 @@
...
+        assert added_service.state == "locked"
+        task.wait()
+        added_service.reread()
+        assert added_service.state == "created"
+
+
+def test_hostprovider_must_be_unlocked_when_his_task_finished(hostprovider: Provider):
+    with allure.step("Run action: action locker and create hostprovider"):
+        task = hostprovider.action_run(name="action-locker")
+    with allure.step("Check if provider is locked and then created"):
```
Comment (dgusakov) [inline tests/functional/test_locked_objects.py:181]:
```suggestion
    with allure.step("Check if provider state is locked and then created"):
```

File: tests/functional/test_locked_objects.py [hunk 5]
```diff
@@ -9,223 +9,195 @@
...
+        hostprovider.reread()
+        assert hostprovider.state == "created"
+
+
+def test_host_and_hostprovider_must_be_unlocked_when_his_task_finished(
+    hostprovider: Provider, host: Host
+):
+    with allure.step("Run action: action locker and create hostprovider"):
+        task = hostprovider.action_run(name="action-locker")
+    with allure.step("Check if provider and host is locked and then created"):
```
Comment (dgusakov) [inline tests/functional/test_locked_objects.py:194]:
```suggestion
    with allure.step("Check if provider and host states are 'locked' and then are 'created'"):
```

File: tests/functional/test_locked_objects.py (hunk 3 above)
Comment (dgusakov) [reply tests/functional/test_locked_objects.py:192]: Pls check allure steps in the whole module.

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: tests/functional/test_locked_objects.py (hunk 1 above)
Comment (vgorkavenko) [reply tests/functional/test_locked_objects.py]: fixed

File: tests/functional/test_locked_objects.py (hunk 2 above)
Comment (vgorkavenko) [reply tests/functional/test_locked_objects.py]: fixed

File: tests/functional/test_locked_objects.py (hunk 3 above)
Comment (vgorkavenko) [reply tests/functional/test_locked_objects.py]: fixed

File: tests/functional/test_locked_objects.py (hunk 4 above)
Comment (vgorkavenko) [reply tests/functional/test_locked_objects.py]: fixed

File: tests/functional/test_locked_objects.py (hunk 5 above)
Comment (vgorkavenko) [reply tests/functional/test_locked_objects.py]: fixed

**Review by dgusakov — APPROVED** [review]

**Pull request merged**